lib/api/kuzzle.js

```diff
@@ -157,6 +161,71 @@ Kuzzle.prototype = new EventEmitter({
 });
 Kuzzle.prototype.constructor = Kuzzle;
 
-module.exports = Kuzzle;
-
+/**
+ * Register handlers and do a kuzzle dump for:
+ * - system signals
+ * - unhandled-rejection
+ * - uncaught-exception
+ */
+function registerErrorHandlers() {
+  var coreDumpSigals = {
+    SIGHUP: 1,
+    // SIGINT: 2,
+    SIGQUIT: 3,
+    // SIGILL: 4, can not be handled
+    SIGABRT: 6,
+    // SIGFPE: 8, can not be handled
+    // SIGKILL: 9, can not be handled
+    // SIGSEGV: 11, can not be handled
+    SIGPIPE: 13,
+    // SIGBUS: 10, can not be handled
+    SIGTERM: 15,
+  };
+  var request = new RequestObject({
+    controller: 'actions',
+    action: 'dump',
+    body: {}
+  });
+
+  process.removeAllListeners('unhandledRejection');
```

Comment:
Why do we remove all these listeners before registering our own? Are listeners pre-registered before Kuzzle starts?

(this question applies to all `removeAllListeners` of this method)